catalog/rocks/validate.go

```diff
@@ -0,0 +1,113 @@
+package rocks
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/ident"
+)
+
+const (
+	MaxPathLength = 1024
+)
+
+var (
+	reValidRef          = regexp.MustCompile(`^[^\s]+$`)
+	reValidBranchID     = regexp.MustCompile(`^\w[-\w]*$`)
+	reValidRepositoryID = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{2,62}$`)
+)
+
+var (
+	ErrInvalidValue = errors.New("invalid value")
+	ErrInvalidType  = errors.New("invalid type")
```

Comment:
I think our errors are not very descriptive, especially if they propagate to the user.
e.g. Running `git commit` without an empty message ends up with:
`Aborting commit due to empty commit message.`
We'll write back:
`message: invalid value`

Reply:
I'll return ErrRequiredValue in case value is required.
The InvalidType are bugs I can fail with panic if we think it will be a better way to handle it. WDYT?

Reply:
Panics are better IMO.